internal/services/containers/kubernetes_cluster_node_pool_resource.go

```diff
@@ -1248,7 +1248,51 @@ func resourceKubernetesClusterNodePoolDelete(d *pluginsdk.ResourceData, meta int
 	return nil
 }
 
-func upgradeSettingsSchema() *pluginsdk.Schema {
+func upgradeSettingsSchemaNodePoolResource() *pluginsdk.Schema {
+	return &pluginsdk.Schema{
+		Type:     pluginsdk.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &pluginsdk.Resource{
+			Schema: map[string]*pluginsdk.Schema{
+				"max_surge": {
+					Type:     pluginsdk.TypeString,
+					Optional: true,
+					// NOTE: O+C because maxSurge must be set to "0" for maxUnavailable to be set
```

Comment:
Is there a difference of the API behavior when this is set to 0 vs this is not set, at the API model? I have concerns that if we do so, there is no way to unset `max_surge` then?

Reply:
The `max_surge` defaults to "10%" by the API even the `upgradeSettings` block is omitted. So exactly one of `maxSurge` and `maxUnavailable` is always set. 

If users want to switch back from `maxUnavailable`, they can add `max_surge="10%"` in the configuration to restore.